prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1120,44 +1120,28 @@ static ssize_t rxr_ep_cancel(fid_t fid_ep, void *context)
  */
 static int efa_set_fi_hmem_p2p_opt(struct efa_ep *efa_ep, int opt)
 {
-	struct efa_hmem_info *hmem_info;
+	int i, ret;
 
-	hmem_info = efa_ep->domain->hmem_info;
-
-	switch (opt) {
+	/* FI_HMEM interfaces to check against */
+	static enum fi_hmem_iface ifaces[] = {
+		FI_HMEM_CUDA,
+		FI_HMEM_NEURON,
+		FI_HMEM_SYNAPSEAI
+	};
+	
 	/*
-	 * TODO: support the other options. We can only support ENABLED
-	 * and PREFERRED when p2p is available. DISABLED is not
-	 * supported yet.
+	 * Check the opt's validity against the first initialized FI_HMEM interface
+	 * encountered from the list above
```

Comment:
Checking against the first is somewhat strange.  Should it be all or any?

Reply:
Compare against the previous logic; this is essentially what was happening before. We would check if `FI_HMEM_CUDA` was initialized (and p2p supported) and set the p2p opt if valid. Otherwise, we would check if `FI_HMEM_NEURON` was intialized and assert the option was `FI_HMEM_P2P_REQUIRED`.

Checking against all is a no-go, since that would restrict valid options to the most restrictive option (`FI_HMEM_P2P_DISABLED`). Checking against any might be a better idea, but what I've implemented is essentially a transposed generic version of what we had before.

Reply:
See another example in our provider [here](https://github.com/ofiwg/libfabric/blob/55e77cda043849f47dcb9780e947b4b28fd5d4de/prov/efa/src/efa_ep.c#L702-L718):

```c
	if (ep->domain->hmem_info[FI_HMEM_CUDA].initialized) {
		/*
		 * Set the default to required. NCCL plugin requires p2p, but
		 * does not call setopt for this option in older NCCL plugin
		 * versions.
		 */
		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
	} else if (ep->domain->hmem_info[FI_HMEM_NEURON].initialized) {
		/* Neuron requires p2p and supports no other modes. */
		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
	} else if (ep->domain->hmem_info[FI_HMEM_SYNAPSEAI].initialized) {
		/* SynapseAI requires p2p and supports no other modes. */
		ep->hmem_p2p_opt = FI_HMEM_P2P_REQUIRED;
	} else {
		/* no hmem devices, disable p2p */
		ep->hmem_p2p_opt = FI_HMEM_P2P_DISABLED;
	}
```

As you can see, we just iterate through an arbitrary set of `FI_HMEM` interfaces and short-circuit upon the first initialized interface.

Reply:
I see.  Related to do we assume we have multiple initialized issue.